### DISCUSSION TITLE
Week 3: Summary of Cafe Menu and Bento Boxes

### DISCUSSION THREAD

**Discussion by bst3ck** [USU-ITLS-x265-23-summer/class-repo#84] [General]
## Week 3 Summary
One aspect of user experience that I have found helps to engage users, is the aspect of gamification. Using games to help teach was a very interesting approach. I actually really enjoyed both the Bento/fruit lesson and the frog/lilypad activities. But I do have to say that the last frog level was pretty difficult. I probably spent more time on that last level than the entire 23 preceding levels. For some reason, I could not remember to use wrap-reverse. But I felt that both the CSS and Flex lessons were very helpful. I could see some very useful cases, but I have to be honest, there were a few I was scratching my head as to why they would need a selector like that vs just writing the HTML in the correct order or building the flex box divs differently.

I am still having a hard time with the FreeCodeCamp. I almost wish we were given a final target image and asked to replicate it. With over 90 steps, the line-by-line entry style feels a little monotonous. But with that being said, I would not have used some of the shortcuts, so it makes sense why they have asked us to follow that pattern. Overall, I enjoyed week 3 and looking forward to next week.

**Comment by metasean** [comment]
>  I could see some very useful cases, but I have to be honest, there were a few I was scratching my head as to why they would need a selector like that vs just writing the HTML in the correct order or building the flex box divs differently.

There are actually a range of reasons, including existing html code (or html code that's being constructed by existing JavaScript) that would be more difficult than just changing selectors, and using the same HTML for a range of different layouts (e.g. desktop, tablet, & mobile). 

> With over 90 steps, the line-by-line entry style feels a little monotonous. 

These platforms evolve over time.  Prior to the first week of this course last summer, there were far less steps to reach the same goal.  However, I 100% with you and really wish it hadn't been changed.  One upshot of this approach is that Web Design section uses exercises that are actually grounded in the bigger picture of a project, instead of being isolated topics. 


> I almost wish we were given a final target image and asked to replicate it.

You should be seeing the final target image, e.g.

https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/7552754/3c40422b-81ed-449a-b0e7-93dab627825b

The challenge with skipping individual steps is that several different topics are covered within each project, and it's really easy to miss an important aspect if you aren't following the actual steps.
